pymeasure/instruments/srs/ldc500series.py

```diff
@@ -0,0 +1,516 @@
+        - TEC: Thermo-electric controller
+    """
+
+    def __init__(self, adapter, name="LDC500Series", **kwargs):
+        super().__init__(adapter, name, **kwargs)
+        self.ld = LDC500SeriesLDSubsystem(self)
+        self.pd = LDC500SeriesPDSubsystem(self)
+        self.tec = LDC500SeriesTECSubsystem(self)
+
+
+class LDC500SeriesLDSubsystem(LDC500SeriesBase):
+    """Subsystem of LDC500Series for control of the laser-diode (LD)."""
+
+    def __init__(self, adapter, name="LDC500Series LD Subsystem", **kwargs):
+        super().__init__(adapter, name, **kwargs)
+
+    interlock_closed = Instrument.measurement(
+        "ILOC?",
+        """Get the status of the interlock, True if closed, False if open.
+        Laser will only operate with a closed interlock.""",
+        values={True: "CLOSED", False: "OPEN"},
+        map_values=True,
+    )
+
+    enabled = Instrument.control(
+        "LDON?",
+        "LDON %s",
+        """Control whether the laser diode current source is enabled (bool).""",
+        validator=strict_discrete_set,
+        values={True: "ON", False: "OFF"},
+        map_values=True,
+    )
+
+    mode = Instrument.control(
+        "SMOD?",
+        "SMOD %s",
+        """Control the laser diode control mode, (str "CC" or "CP").
+
+        If the laser is on when ``mode`` is changed, the controller performs a *bumpless
+        transfer* to switch control modes on-the-fly.
+
+        If ``mode`` is changed from "CC" to "CP", the present value of the photodiode current
+        (or equivalently optical power) is measured and the CP setpoint is set to this measurement.
+        If the measured photodiode current or power exceed the setpoint limit, an error is thrown.
+
+        If the ``mode`` is changed from "CP" to "CC", the present value of the laser current is
+        measured and the CC setpoint set to this measurement. Note that, by hardware design, the
+        measured laser current can never exceed the setpoint limit.
+        """,
+        validator=strict_discrete_set,
+        values=["CC", "CP"],
+        check_set_errors=True,
+    )
+
+    # --- direct ld current control ---
+
+    current = Instrument.measurement(
+        "RILD?",
+        """Measure the laser diode current, in mA (float).""",
+    )
+
+    current_setpoint = Instrument.control(
+        "SILD?",
+        "SILD %g",
+        """Control the laser diode current setpoint, in mA, when ``mode == "CC"`` (float).""",
+        check_set_errors=True,
+    )
+
+    current_limit = Instrument.control(
+        "SILM?",
+        "SILM %g",
+        """Control the laser diode current limit, in mA (float).
+
+        The ouput current is clamped to never exceed ``current_limit`` under any conditions.
+        If ``current_limit`` is reduced below ``current_setpoint``,
+        ``current_setpoint`` is "dragged" down with ``current_limit``.
+        """,
+        check_set_errors=True,
+    )
+
+    current_range = Instrument.control(
```

Comment:
here also, `current_range_high_enabled` could give you a boolean, which is easier than a string property.